### ISSUE TITLE
android wallet resyncing from scratch even if already synced

### ISSUE THREAD

**Issue by lsm5** [cake-tech/cake_wallet#1213]
My android wallet had finished syncing yesterday and it said `Synchronized`. But today it started syncing again on its own with 2 million blocks to go. I have downloaded the app from github releases.

If it matters I've set background sync mode to `Aggressive`.

**Comment by lsm5** [comment]
and now it shows balance 0.0 even though it says `Synchronized`.

**Comment by nahuhh** [comment]
1. Turn off background sync (settings > connection)
2. Rescan blockchain using a restore date from before your first transaction (settings > connection > rescan)
3. see if the problem goes away

**Comment by kinghat** [comment]
using github releases this has been happening the last two releases as well.

**Comment by nahuhh** [comment]
Follow suggestion, report back

**Comment by nahuhh** [comment]
1. Turn off background sync

**Comment by tuxpizza** [comment]
This is new to me. I don't think I've ran into this issue and I've used both Unobtrusive and Aggressive background sync for months. Could be an issue with the OS killing CW sooner than expected. I'll open this internally, thanks for the report. 

And for reference what version of Android are you guys using?

**Comment by nahuhh** [comment]
Just lost entire wallet history including sub accounts

usually the bug syncs from genesis, this time it showed synchronized as if i entered "today" and rescanned. 0 transaction history

**Comment by belst-n** [comment]
Same here...

**Comment by tuxpizza** [comment]
Temporary fix in progress for this! Until we can overhaul the background sync.

**Comment by nahuhh** [comment]
just posting to share my pain. 

![sr3vbqvfg4r9tswa.jpg](https://github.com/cake-tech/cake_wallet/assets/50635951/9f37bc18-1401-46a3-bf17-1e35845f965d)

**Comment by tuxpizza** [comment]
@nahuhh Just making sure.... is this with background sync on? Or are you running into this with it off also?

**Comment by nahuhh** [comment]
That's with it on.
Left it on to diagnose the issue.

another related issue i observed:
it ONCE loaded the wrong wallet cache when "sync all" was enabled. 

meaning, the transactions screen showed tge transactions of another wallet, and the sync bar kept flashing 2 different "blocks remaining".

(ive turned it off since your last comment)

**Comment by tuxpizza** [comment]
Hey all. Cake Wallet v4.13.3 and Monero.com v1.10.3 has been released and includes fixes for background sync and internal backups incase of wallet corruption. Now when you turn Background sync on, it will ask you to Allow Cake Wallet to run in unrestricted battery mode, which will prevent the OS from force closing Cake in the background, and allow background sync to work better. This is optional, but recommended. These two things in combination should prevent any wallet corruption, and if there is it will restore last known good wallet open.

**Comment by TollgateTrash** [comment]
Getting same error, have synced my wallet twice now and it's in the third. Starts showing balances and everything looks good. Gets down to the last 180 blocks are so, slows WAY down and then syncs and shows balance at zero and shows no transactions excepts one changelly trade done years ago still chills in transaction page. I just turned off background sync and am trying to restore the wallet to a different, clean phone....

**Comment by nahuhh** [comment]
Please 

1. Ship it disabled
2. remove it

i prefer option 2. Thia feature hurts the brand

**Comment by nahuhh** [comment]
> Please 
> 
> 1. Ship it disabled
> 2. remove it
> 
> i prefer option 2. Thia feature hurts the brand

pinging @OmarHatem28 @tuxpizza ! Enough is enough

**Comment by nahuhh** [comment]
I personally havent used cake in > 1 month. After > 1 yr, I'm no longer interested in resyncing weekly.

The feature has ALWAYS been broken

**Comment by nahuhh** [comment]
It shouldnt take this long. 💢

**Comment by tuxpizza** [comment]
We've been debating removing it but we recently decided to add a warning message due to its experimental nature, which will be in next update

**Comment by nahuhh** [comment]
It was enabled by default and persists.
its broken on android, not "experimental".

risk vs reward. 
useless and insecure feature. Kill it

**Comment by tuxpizza** [comment]
It's not enabled by default. Only reason it would be enabled is if you were restoring from a backup that had it enabled.

**Comment by nahuhh** [comment]
experimental (broken) = dont push it to release channels...

instead, users can leave the app running in recents.

Ios doesnt appear to have the same issue, butbthis is literally expensive AND leaves your spend keys unlocked (crazy bs feature)

**Comment by nahuhh** [comment]
Yknow how many other wallets leave your spend keys unlocked, call home, and corrupt wallets as features?  

yeah

**Comment by tuxpizza** [comment]
I have mentioned these complaints to our main dev and he agrees to disable background sync for now until we can implement it better.

Sorry for the frustration it has caused. This feature was definitely NOT released "experimental" like I said in the previous comment, I was only mentioning that in relation to what we know now. But clearly it had unknown issues that were discovered after it was released and we already tried to fix it the best we could, but since these patches have run their course and been proven not to ultimately fix it we are disabling it in the latest update.
